src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxButton.java

```diff
@@ -228,6 +249,16 @@ protected void doRendererPaint(final Graphics g, final ButtonModel buttonModel,
 
         rendererPane.paintComponent(g, c, this, left, top, cWidth, height, shouldValidate); // h - (insets.top + insets.bottom) );
 
+        // set the accessible name to the displayed text in JComboBox.
+        // screen magnifier queries to get the accessible name to display magnified text.
```

Comment:
I still do not think that setting accessible name in paint is a good idea. Setting it during the initialization and adding a property change listener to the accessible context to track changes seems like a better way to do it.

Reply:
Ok, I will check this.

Reply:
> I still do not think that setting accessible name in paint is a good idea.

It's been my biggest concern. But I didn't even think about it this way.

The problem is that the displayed value is known only when the component is painted. Other components seem to handle it… somehow.

Reply:
> Setting it during the initialization and adding a property change listener to the accessible context to track changes seems like a better way to do it.

@azuev-java We have a ItemListener for JComboBox defined in AquaComboBoxUI here to handle any changes in combobox selected item.

https://github.com/openjdk/jdk/blob/9e6cb620486ac7b0adaefeb2000babf3ea31207f/src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java#L135

Tried setting the `accessibleName` for arrowButton in `itemStateChanged` method. So whenever combobox value  changes using list selection, **accessibleName is set to the selected Item**.

For initial accessibleName, it can be set to the first value of data model inside `AquaComboBoxButton constructor` code.

`However, this doesn't work with custom renederer.`

Reply:
@azuev-java Please review.